File: src_local/utils/memory_pruning_agent.py
import sqlite3
import logging
from pathlib import Path

logger = logging.getLogger(__name__)

class MemoryPruningAgent:
    """Agente responsável por evitar a inflação de dados e manter o sistema ágil."""
    
    def __init__(self, project_root):
        self.db_path = Path(project_root) / "system_vault.db"

    def prune_old_logs(self, days=90):
        """Remove registros do histórico mais antigos que X dias."""
        import time
        from src_local.utils.logging_config import log_performance
        start_t = time.time()
        
        logger.info(f"🧠 [Pruning] Iniciando limpeza de registros antigos (> {days} dias)...")
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            cursor.execute(f"DELETE FROM health_history WHERE timestamp < date('now', '-{days} days')")
            deleted = cursor.rowcount
            conn.commit()
            conn.close()
            if deleted > 0:
                logger.info(f"✨ [Pruning] {deleted} registros comprimidos.")
            
            log_performance(logger, start_t, "🧠 [Pruning] Database maintenance")
        except Exception as e:
            logger.error(f"❌ Erro na poda de memória: {e}")

File: src_local/utils/memory_pruning_agent.ts
import { Database } from "bun:sqlite";
import winston from "winston";
import { Path } from "../core/path_utils.ts";

const logger = winston.child({ module: "MemoryPruningAgent" });

/**
 * 🧠 Agente de Poda de Memória (Bun Version).
 * Responsável por evitar a inflação de dados e manter o sistema ágil,
 * removendo registros antigos do banco de dados SQLite.
 */
export class MemoryPruningAgent {
    private dbPath: string;

    constructor(projectRoot: string) {
        this.dbPath = new Path(projectRoot).join("system_vault.db").toString();
    }

    /**
     * Remove registros do histórico mais antigos que X dias.
     */
    pruneOldLogs(days: number = 90): void {
        const startT = Date.now();
        logger.info(`🧠 [Pruning] Iniciando limpeza de registros antigos (> ${days} dias)...`);

        try {
            const db = new Database(this.dbPath);

            // Note: date('now') in SQLite returns UTC date.
            const query = db.query(`DELETE FROM health_history WHERE timestamp < datetime('now', '-${days} days')`);
            query.run();

            // Bun:sqlite doesn't have rowcount directly in run() for DELETE, 
            // but we can assume success if no error.

            db.close();

            const duration = (Date.now() - startT) / 1000;
            logger.info(`✨ [Pruning] Database maintenance concluída em ${duration.toFixed(4)}s.`);
        } catch (e) {
            logger.error(`❌ Erro na poda de memória: ${e}`);
        }
    }
}
